### PR TITLE
LPS-168124 Add new methods to ServiceTrackerList to get list and array

### PR DESCRIPTION
**Pull request by W2jx**

### CODE REVIEW

File: modules/core/osgi-service-tracker-collections/src/main/java/com/liferay/osgi/service/tracker/collections/internal/list/ServiceTrackerListImpl.java [hunk 1]
```diff
@@ -81,6 +82,22 @@ public int size() {
 		return _services.size();
 	}
 
+	@Override
+	public <E> E[] toArray(E[] array) {
+		return toList().toArray(array);
```
Comment (dantewang) [inline modules/core/osgi-service-tracker-collections/src/main/java/com/liferay/osgi/service/tracker/collections/internal/list/ServiceTrackerListImpl.java:87]: @W2jx Please avoid method chaining here.

File: modules/core/osgi-service-tracker-collections-test/src/testIntegration/java/com/liferay/osgi/service/tracker/collections/list/test/ServiceTrackerListTest.java [hunk 2]
```diff
@@ -69,6 +70,33 @@ public void tearDown() {
...
+			Collection<ServiceRegistration<TrackedOne>> serviceRegistrations =
+				registerServices(TrackedOne.class, trackedOnes);
+
+			List<TrackedOne> trackedOneList = serviceTrackerList.toList();
+
+			Assert.assertArrayEquals(
+				serviceTrackerList.toString(), trackedOnes,
+				trackedOneList.toArray(new TrackedOne[0]));
+
+			TrackedOne[] trackedOnes1 = serviceTrackerList.toArray(
```
Comment (dantewang) [inline modules/core/osgi-service-tracker-collections-test/src/testIntegration/java/com/liferay/osgi/service/tracker/collections/list/test/ServiceTrackerListTest.java:90]: @W2jx When there's 1, there's 2. Please rename this, maybe call it `trackedOneArray`.

**Review by dantewang — CHANGES_REQUESTED** [review]
@W2jx Please see my comments and modify. Also, the example Brian gave is not modified in your `apply`, please double check whether you've applied to all the cases.

File: modules/core/osgi-service-tracker-collections-test/src/testIntegration/java/com/liferay/osgi/service/tracker/collections/list/test/ServiceTrackerListTest.java (hunk 2 above)
Comment (W2jx) [reply modules/core/osgi-service-tracker-collections-test/src/testIntegration/java/com/liferay/osgi/service/tracker/collections/list/test/ServiceTrackerListTest.java:90]: I will fix them.

**Comment by W2jx** [comment]
Hi @dantewang , I have rebased, rechecked the code and I have fixed the problem mentioned in the comments.

**Comment by W2jx** [comment]
ci:test:relevant

**Comment by W2jx** [comment]
ci:test:sf

**Comment by catarineaguiar** [comment]
@W2jx no regressions found.

**Comment by dantewang** [comment]
@W2jx See https://github.com/liferay-core-infra/liferay-portal/pull/2437

https://github.com/liferay-core-infra/liferay-portal/pull/2437/commits/c9dddcdcebab0486edcd6ca1214f9df38cf00932 & https://github.com/liferay-core-infra/liferay-portal/pull/2437/commits/2a41d3ac22685a24105f1a9bf2450a8432e33377

**Pull request closed without merging**